File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/proc/AttProcessor.java
package ru.citeck.ecos.records3.record.op.atts.service.proc;

import org.jetbrains.annotations.NotNull;
import ru.citeck.ecos.commons.data.DataValue;
import ru.citeck.ecos.commons.data.ObjectData;

import java.util.Collection;
import java.util.Collections;
import java.util.List;

public interface AttProcessor {

    @NotNull
    Object process(@NotNull ObjectData attributes, @NotNull DataValue value, @NotNull List<DataValue> arguments);

    @NotNull
    String getType();

    @NotNull
    default Collection<String> getAttributesToLoad(@NotNull List<DataValue> arguments) {
        return Collections.emptySet();
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/proc/AttProcessor.kt
package ru.citeck.ecos.records3.record.op.atts.service.proc

import ru.citeck.ecos.commons.data.DataValue
import ru.citeck.ecos.commons.data.ObjectData

interface AttProcessor {

    fun process(attributes: ObjectData, value: DataValue, args: List<DataValue>): Any?

    fun getAttsToLoad(arguments: List<DataValue>): Collection<String> = emptySet()

    fun getType(): String
}
